File: python/grains/grains.py
TOTAL_SQUARES = 64


def square(number):
    if number in range(1, TOTAL_SQUARES + 1):
        return 2 ** (number - 1)
    else:
        raise ValueError("error")


def total(number):
    if number in range(1, TOTAL_SQUARES + 1):
        return (2 ** TOTAL_SQUARES) - 1
    else:
        raise ValueError("error")


def version():
    VERSION = 1

File: go/grains/grains.go
package grains

import "errors"

const Version = "1"

const maxSquares int = 64

func Square(n int) (uint64, error) {
	if n < 1 || n > maxSquares {
		return uint64(0), errors.New("Invalid")
	}

	return uint64(1 << uint(n-1)), nil
}

func Total() uint64 {
	return uint64((1 << maxSquares) - 1)
}
